### PR TITLE
backend/feat/sendSms-api-for-provider-dashboard

### PR DESCRIPTION
**Pull request by Anuragini-2506**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [x] This PR contains API breaking changes

## How did you test it?

tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs [hunk 1]
```diff
@@ -549,3 +556,10 @@ updateSubscriptionDriverFeeAndInvoice merchantShortId apiTokenInfo driverId req
   transaction <- buildTransaction Common.UpdateSubscriptionDriverFeeAndInvoiceEndpoint apiTokenInfo driverId $ Just req
   T.withTransactionStoring transaction $
     Client.callDriverOfferBPP checkedMerchantId (.drivers.updateSubscriptionDriverFeeAndInvoice) driverId req
+
+sendSMSToDriverViaDashboard :: ShortId DM.Merchant -> ApiTokenInfo -> Id Common.Driver -> DDriver.SendSmsReq -> FlowHandler APISuccess
+sendSMSToDriverViaDashboard merchantShortId apiTokenInfo driverId req = withFlowHandlerAPI $ do
+  checkedMerchantId <- merchantAccessCheck merchantShortId apiTokenInfo.merchant.shortId
+  transaction <- buildTransaction Common.SendSmsToDriverViaDashboardEndPoint apiTokenInfo driverId (Just $ DDriver.VolunteerTransactionStorageReq apiTokenInfo.personId.getId driverId.getId (show req.messageKey))
```
Comment (RitikaHanish) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs:563]: Why are we passing redundant data here ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Driver.hs [hunk 2]
```diff
@@ -160,6 +161,22 @@ type FleetStatsAPI =
     :> "fleet"
     :> Get '[JSON] Common.FleetStatsRes
 
+-------------------------------------------------------------------
+------- Send Sms to Driver ----------------------------------------
+
+type SendSmsToDriverViaDashboardAPI =
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Driver.hs:167]: What's the difference In SendSmsToDriverViaDashboardAPI and SendSmsToDriverAPI ?

Comment (Anuragini-2506) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Driver.hs:167]: So In this dashboard use case, we need to track the volunteer sms hit count. So we require to increase the hit limit whenever the voluteer is able to successfully send the message.

But increasing the count is done in driver-app and driver app does not get the volunteer data. So,

for dashboard side following api is exponsed -` SendSmsToDriverViaDashboardAPI`
```
  Capture "driverId" (Id Common.Driver)
    :> "sendSms"
    :> ReqBody '[JSON] DDriver.SendSmsReq
    :> Post '[JSON] APISuccess
```

and for driver app side, as we require volunteer id as well, I am passing volunteer id from the apiTokenInfo internall by hitting this internal dashboard endpoint for driver-app.
 
```
type SendSmsToDriverAPI =
  Capture "driverId" (Id Common.Driver)
    :> Capture "volunteerId" Text
    :> "sendSms"
    :> ReqBody '[JSON] DDriver.SendSmsReq
    :> Post '[JSON] APISuccess
```

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/MessageBuilder.hs [hunk 3]
```diff
@@ -155,3 +157,18 @@ buildCollectCashMessage merchantId req = do
...
     merchantMessage.message
       & T.replace (templateText "amount") req.amount
+
+data BuildGenericMessageReq = BuildGenericMessageReq {}
+  deriving (Generic)
+
+buildGenericMessage :: (EsqDBFlow m r, CacheFlow m r) => Id DM.Merchant -> DMM.MessageKey -> BuildGenericMessageReq -> m Text
+buildGenericMessage merchantId messageKey _ = do
+  merchantMessage <-
+    QMM.findByMerchantIdAndMessageKey merchantId messageKey
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/MessageBuilder.hs:167]: is it cached, if not then can we run this in replica ?

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs (hunk 1 above)
Comment (Anuragini-2506) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs:563]: @SrinivasNaikB  needed this data for analysis purpose. So I am storing the 
* volunteerId
* driverId 
* messageKey 

in the request of the transaction.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/MessageBuilder.hs (hunk 3 above)
Comment (Anuragini-2506) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/MessageBuilder.hs:167]: `import qualified Storage.CachedQueries.Merchant.MerchantMessage as QMM`

( It is a cached query. In this file, the qualified import is named as `QMM`

**Pull request closed without merging**